src/libertem/udf/holography.py

```diff
@@ -0,0 +1,199 @@
+        """
+        extra_shape = self.params.out_shape
+        if not self.params.precision:
+            dtype = np.complex64
+        else:
+            dtype = np.complex128
+        return {
+            "wave": self.buffer(kind="nav", dtype=dtype, extra_shape=extra_shape)
+        }
+
+    def get_task_data(self):
+        """
+        Updates `task_data`
+        Returns
+        -------
+        kwargs : dict
+            A dictionary with kwargs:
+            kwargs['aperture'] : array-like
+                Side band filter aperture (mask)
+            kwargs['slice'] : slice
+                Slice for slicing FFT of the hologram
+        """
+
+        out_shape = self.params.out_shape
+        sy, sx = self.meta.partition_shape.sig
+        oy, ox = out_shape
+        f_sampling = (1. / oy, 1. / ox)
+        sb_size = self.params.sb_size * np.mean(f_sampling)
+        sb_smoothness = sb_size * self.params.sb_smoothness * np.mean(f_sampling)
+
+        f_freq = freq_array(out_shape)
+        aperture = aperture_function(f_freq, sb_size, sb_smoothness)
+
+        y_min = int(sy / 2 - oy / 2)
+        y_max = int(sy / 2 + oy / 2)
+        x_min = int(sx / 2 - ox / 2)
+        x_max = int(sx / 2 + oy / 2)
+        slice_fft = ([slice(y_min, y_max), slice(x_min, x_max)])
+
+        kwargs = {
+            'aperture': aperture,
+            'slice': slice_fft
+        }
+        return kwargs
+
+    def process_frame(self, frame):
+        """
+        Reconstructs holograms outputting results into 'wave'
+
+        Parameters
+        ----------
+        frame
+           single frame (hologram) of the data
+        """
+        if not self.params.precision:
+            frame = frame.astype(np.float32)
+        # size_x, size_y = self.params.out_shape
+        frame_size = self.meta.partition_shape.sig
+        sb_pos = self.params.sb_position
+        aperture = self.task_data.aperture
+        slice_fft = self.task_data.slice
+
+        fft_frame = fft2(frame) / np.prod(frame_size)
+        fft_frame = np.roll(fft_frame, sb_pos[0], axis=0)
+        fft_frame = np.roll(fft_frame, sb_pos[1], axis=1)
+
+        fft_frame = fftshift(fftshift(fft_frame)[slice_fft])
+
+        fft_frame = fft_frame * aperture
+
+        wav = ifft2(fft_frame) * np.prod(frame_size)
+        self.results.wave[:] = wav
+
+
+def reconstruct_holograms(ctx, dataset,
+                          out_shape,
+                          sb_position,
+                          sb_size,
+                          sb_smoothness=0.05,
+                          precision=True):
+    """
+    Reconstruct hologram using standard Fourier method
+
+    Parameters
+    ----------
+    ctx
+        Context class that contains methods for loading datasets, creating jobs on them
+        and running them
+
+    dataset
+        dataset to work on
+
+    out_shape : (int, int)
+        Shape of the outputted complex wave image
```

Comment:
How does one choose the right output shape? What does that depend on?

Reply:
That's a bit flexible. It depends on applications and parameters. My general rule that for low magnification holography it doesn't make sense to have images larger than diameter of sideband. But for high resolution holography people use the output of the same size or binning 2...